Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -34,6 +54,39 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 		}
 
 		ResourceDictionary _mergedInstance;
+		ICollection<ResourceDictionary> _mergedDictionaries;
+		public ICollection<ResourceDictionary> MergedDictionaries { get { return _mergedDictionaries; } }
+
+		public ResourceDictionary()
+		{
+			var collection = new ObservableCollection<ResourceDictionary>();
+			collection.CollectionChanged += MergedDictionaries_CollectionChanged;
+			_mergedDictionaries = collection;
+		}
+
+		private void MergedDictionaries_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		{
+			// New Items
+			var newItems = e.NewItems?.Cast<ResourceDictionary>();
+			if (newItems != null)
+			{
+				foreach (var item in newItems)
+					item.ValuesChanged += Item_ValuesChanged;
+
+				if (newItems.Count() > 0)
+					OnValuesChanged(newItems.SelectMany(x => x).ToArray());
+			}
+			// Old Items
+			var oldItems = e.OldItems?.Cast<ResourceDictionary>();
+			if (oldItems != null)
+				foreach (var item in oldItems)
+					item.ValuesChanged -= Item_ValuesChanged;
```

Comment:
Switching on e.Action doesn't change anything. My current code works for Add/Remove/Replace/Move, without any switching, as they come through with either NewItems or OldItems. The Reset action that comes through will have blank e.OldItems, hence there is no way to remove the event handler, for all items that have been cleared. That's the only issue with this piece of code, I can't get access to the items removed, unless I use one of the options I mentioned above.